pkg/providers/tinkerbell/hardware/etcd.go

```diff
@@ -82,3 +78,19 @@ func (er *ETCDReader) getUnprovisionedTinkerbellHardware(ctx context.Context) ([
 
 	return selectedHardware.Items, nil
 }
+
+// RufioMachinesFromEtcd fetches rufio machine objects from the cluster and inserts into catalogue.
+func (er *ETCDReader) RufioMachinesFromEtcd(ctx context.Context) error {
```

Comment:
Do we need to append `FromEtcd` to the methods? 

Side note: this reads from Kube API Server as opposed to etcd. We should consider renaming to something a little more appropriate.

Reply:
As discussed I have renamed the struct and the methods 👍